### PR TITLE
feat: use window location to open Okta Verify

### PR DESCRIPTION
**Pull request by jhe-okta**
- Use window.location to open custom URI to launch app from browser.
- Update the mock where custom URI is with non-http scheme so that we can mimic the flow in test. Otherwise we will have to mock the http request then the UI will be updated so we cannot test the UI.

OKTA-279056

### CODE REVIEW

File: test/testcafe/spec/DeviceChallengePollViewOktaVerify_spec.js [hunk 1]
```diff
@@ -1,82 +1,35 @@
...
-import identify from '../../../playground/mocks/idp/idx/data/identify';
-import identifyWithDeviceProbingLoopback from '../../../playground/mocks/idp/idx/data/identify-with-device-probing-loopback';
-import loopbackChallengeNotReceived from '../../../playground/mocks/idp/idx/data/identify-with-device-probing-loopback-challenge-not-received';
 import identifyWithLaunchAuthenticator from '../../../playground/mocks/idp/idx/data/identify-with-device-launch-authenticator';
 
 const logger = RequestLogger(/introspect|probe|cancel|launch|launch-okta-verify|poll/);
 
+// const identifyWithLaunchAuthenticatorHttpCustomUri = Object.assign({}, identifyWithLaunchAuthenticator.default);
+const identifyWithLaunchAuthenticatorHttpCustomUri = JSON.parse(JSON.stringify(identifyWithLaunchAuthenticator));
+debugger;
```
Comment (nikhilvenkatraman-okta) [inline test/testcafe/spec/DeviceChallengePollViewOktaVerify_spec.js]: remove debugger statement.

File: test/testcafe/spec/DeviceChallengePollViewOktaVerify_spec.js [hunk 2]
```diff
@@ -1,82 +1,35 @@
...
+const mockHttpCustomUri = 'http://localhost:3000/launch-okta-verify';
+// replace custom URI with http URL so that we can mock and verify
+identifyWithLaunchAuthenticatorHttpCustomUri.authenticatorChallenge.value.href = mockHttpCustomUri;
```
Comment (nikhilvenkatraman-okta) [inline test/testcafe/spec/DeviceChallengePollViewOktaVerify_spec.js]: can you add a comment explaining, why we are setting the href here in the test and not in the mock file ?

File: test/testcafe/spec/DeviceChallengePollViewOktaVerify_spec.js [hunk 3]
```diff
@@ -1,82 +1,35 @@
...
 import DeviceChallengePollPageObject from '../framework/page-objects/DeviceChallengePollPageObject';
-import IdentityPageObject from '../framework/page-objects/IdentityPageObject';
-import identify from '../../../playground/mocks/idp/idx/data/identify';
-import identifyWithDeviceProbingLoopback from '../../../playground/mocks/idp/idx/data/identify-with-device-probing-loopback';
-import loopbackChallengeNotReceived from '../../../playground/mocks/idp/idx/data/identify-with-device-probing-loopback-challenge-not-received';
 import identifyWithLaunchAuthenticator from '../../../playground/mocks/idp/idx/data/identify-with-device-launch-authenticator';
 
 const logger = RequestLogger(/introspect|probe|cancel|launch|launch-okta-verify|poll/);
 
+// const identifyWithLaunchAuthenticatorHttpCustomUri = Object.assign({}, identifyWithLaunchAuthenticator.default);
```
Comment (nikhilvenkatraman-okta) [inline test/testcafe/spec/DeviceChallengePollViewOktaVerify_spec.js]: remove commented out code if not used.

File: test/testcafe/spec/DeviceChallengePollViewOktaVerify_spec.js (hunk 2 above)
Comment (jhe-okta) [reply test/testcafe/spec/DeviceChallengePollViewOktaVerify_spec.js]: There's a comment on line 11. Replacing a custom URI with a HTTP URL so we can mock it and test it.

**Pull request closed without merging**